### ISSUE TITLE
Milestone 5 Task 5: Adapt the current codebase

### ISSUE THREAD

**Issue by FLX-20** [FLX-20/Data-Science-Toolkits-and-Architectures-group-1#59]
Make the required adaptations in the existing backend codebase for the milestone 5. This includes the following tasks.
- adapt the code to train and test the CNN on the MNIST dataset.
- train and evaluate various models, defined in a JSON file, using "Weights & Biases" to identify the best model.
- save the best model in the model folder for the Flask front end.

**Comment by FLX-20** [comment]
What is your current status of this issue? Are there problems?

**Comment by FLX-20** [comment]
Could you try to finish this part/issue by the 8th of January to ensure we have enough time to bring everything together?

**Comment by Tjerk8389** [comment]
Do you want that both of us perform this task, as we are both assigned to do to this task.

**Comment by FLX-20** [comment]
Yes, I thought it would be easier for you to share information, because various things in the `/src` need to be adjusted.

**Comment by Tjerk8389** [comment]
Just for clarification, I need to do these tasks in a new terminal right? Not in the terminal/branches which are connected to changes in Git (Github).

**Comment by FLX-20** [comment]
You have to create a new `main-dev-5-branch` switch to this branch and in this branch create a new branch, for example `mnist_adjustment`. In this way you are consistent with the current branching strategy. In the end switch into your new branch `mnist_adjustment` and make your changes in this branch.

**Comment by Tjerk8389** [comment]
In which branch do I need to create the main-dev-5-branch, because I made it in the report-5-chapter-2 branch and I think this is incorrect. Therefore I delete the main-dev-5-branch, and create a new one in another branch. But which branch do I need to make the main-dev-5-branch then, in the main-report-5-branch or from the main branch?

**Comment by FLX-20** [comment]
You need to create the main-dev-5-branch in the main branch.

**Comment by Tjerk8389** [comment]
I have created the main-dev-5-branch in the main branch but I had a question because I need to adapt the code from files in the src directory. And if I do that are you also going to merge those changes into the main branch? Isn't that going to change other files that are necessary for the final version of the milestone?

**Comment by FLX-20** [comment]
In the main-dev-5-branch create another new branch where you make your changes.
You can change everything in the src files, becuase I excluded the flask application into a new directory called app on my branch.

**Comment by Tjerk8389** [comment]
I made a new branch where I am going to report my changes. I hope I can finish it today

**Comment by FLX-20** [comment]
That would be nice. 

Just to led you know. I won't be available from the 14th of January. So I am looking forward to finish the last milestone until this date. I think this should not be a problem.

**Comment by Tjerk8389** [comment]
I have made changes in the cnn_operations.py, train.py, evaluate.py, save_load_models.py files in the 'mnist_adjustments' branch. I hope it's according to your preferences.

**Comment by FLX-20** [comment]
You forgot this push it. It is not on the Github repo.

**Comment by Tjerk8389** [comment]
Now I have pushed it to the Github repo

**Comment by FLX-20** [comment]
Ok thank you for your work, I will integrate this code into the codebase and bring the milestone to the end.

**Comment by FLX-20** [comment]
I finished your task, adapted the frontend to the backend, and finally built the release.
So, the project is officially finished.
